exec/supervisor/config.go

```diff
@@ -0,0 +1,80 @@
+// Copyright (c) 2019 Cisco and/or its affiliates.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at:
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package supervisor
+
+import "github.com/pkg/errors"
+
+// Config represents configuration file for the supervisor procedure
+type Config struct {
+	// The configuration file consists from a list of processes
+	Processes []Process `json:"processes"`
+}
+
+// Process is a single process instance started by the supervisor
+type Process struct {
+	// Unique name of the process
+	Name string `json:"name"`
+
+	// File where the log output will be written
+	LogFilePath string `json:"log-file-path"`
+
+	// Path to the process binary file
+	BinaryPath string `json:"binary-path"`
+
+	// A list of arguments the process will be started with
+	Args []string `json:"args"`
+
+	// Marks process as required. A required process triggers termination
+	// or all other processes known to supervisor
+	Required bool `json:"required"`
+
+	// Hooks allow to run custom commands/events at process termination
+	Hooks []Hook `json:"hooks"`
```

Comment:
In [the previous review comment](https://github.com/ligato/cn-infra/pull/393/#discussion_r283325304), I actually proposed to add `Hooks` to the `Config` type, instead of `Process`. 

This allows to have single global hook that receives events for all programs. The event info can be passed to the hook command using env vars:

- `SUPERVISOR_EVENT_TYPE` - represents type of hook event; for now only one type: `process-state` ([more](http://supervisord.org/events.html#event-types) could be added later)
- `SUPERVISOR_PROCESS_NAME` - value of `Process.Name`
- `SUPERVISOR_PROCESS_STATE` - value of process status coming from notification channel

These are enough for handling process state changes for any process. We can add more later when needed.